ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/PubsubMessageToObjectNode.java

```diff
@@ -456,10 +453,12 @@ private void expandMapType(String jsonFieldName, ObjectNode value, Field field,
 
       final ArrayNode unmapped = Json.createArrayNode();
       value.fields().forEachRemaining(e -> {
-        ObjectNode kv = Json.createObjectNode().put(FieldName.KEY, e.getKey());
+        ObjectNode kv = Json.createObjectNode();
         valueFieldOption
             .ifPresent(valueField -> processField(e.getKey(), valueField, e.getValue(), kv, props));
-        unmapped.add(kv);
+        // add key after processField so it can't be dropped due to e.getKey() matching
+        // FieldName.KEY when e.getValue() is null or empty or can't be coerced
+        unmapped.add(kv.put(FieldName.KEY, e.getKey()));
```

Comment:
previously this wouldn't remove data from `kv` because `valueField.getName()` would never match `FieldName.KEY`, but now we have to make sure `kv` is empty when passed to `processField` because `e.getKey()` could be anything.